### ISSUE TITLE
[Mac] Delay between crc setup and actual daemon start

### ISSUE THREAD

**Issue by praveenkumar** [code-ready/crc#3148]
Using launchd we start `crc daemon` process as part of setup but there is bit delay around daemon process is actually running and serving to the requests so in scripted way user might face issues.
```
 ./crc setup && ./crc start
[...]
INFO Checking if crc daemon plist file is present and loaded 
INFO Adding crc daemon plist file and loading it  
Your system is correctly setup for using CRC. Use 'crc start' to start the instance
Is 'crc daemon' running? Cannot reach daemon API: Get "http://unix/api/version": dial unix /Users/prkumar/.crc/crc-http.sock: connect: connection refused
```

```
./crc setup && sleep 10 && ./crc start
[...]
INFO Checking if crc daemon plist file is present and loaded 
INFO Adding crc daemon plist file and loading it  
Your system is correctly setup for using CRC. Use 'crc start' to start the instance
INFO Checking if running as non-root              
INFO Checking if crc-admin-helper executable is cached 
INFO Checking for obsolete admin-helper executable 
INFO Checking if running on a supported CPU architecture 
[...]
```

At the end of setup we should have a check for daemon availability to fulfill the requests.

**Comment by adrianriobo** [comment]
This behavior should be fixed based on #2687 with #2722

**Comment by praveenkumar** [comment]
> This behavior should be fixed based on #2687 with #2722

Both were the tray issue.

**Comment by adrianriobo** [comment]
Well the issue was solved adding the check on the tray but the #2687 was not talking about the tray it said `setup` does not guarantee the daemon valid state..

Now you are properly extended the check :+1:
